### PR TITLE
`InlineKeyboardButton` cleanup

### PR DESCRIPTION
**Pull request by WaffleLapkin**
- Make `InlineKeyboardButtonKind::Pay.0` have type `True`
- Add more constructors to `InlineKeyboardButton`
- Make code in `inline_keyboard_button.rs` nicer
- Remove useless `InlineKeyboardButton::{text, kind}`
- Document `InlineKeyboardButton` constructors
- Update changelog

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -7,6 +7,20 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
...
+
+- `InlineKeyboardButton::{pay, login, web_app, callback_game, pay}` constructors ([#231][pr231])
+
+### Changed
+
+-  `InlineKeyboardButtonKind::Pay`'s only field now has type `True` ([#231][pr231])
+
+### Removed
+
+- `InlineKeyboardButton::{text, kind}` functions ([#231][pr231])
```
Comment (Hirrolot) [inline CHANGELOG.md:20]: Maybe just deprecate them instead of hard removing?

Comment (WaffleLapkin) [reply CHANGELOG.md:20]: IMO they are useless (you need to set both kind and text on creation), but I'm fine with deprecating them

Comment (Hirrolot) [reply CHANGELOG.md:20]: I'm just afraid that someone's already using these functions, I guess it's better to deprecate them.

Comment (WaffleLapkin) [reply CHANGELOG.md:20]: Sure 👍🏻

Comment (WaffleLapkin) [reply CHANGELOG.md:20]: Done

**Review by Hirrolot — APPROVED** [review]

**Pull request merged**